core/data/fetch.py

```diff
@@ -125,6 +130,54 @@ def img_path_from_row(row: pd.Series, index: int, column="identifier", extra=Non
         if extra:
             out += extra
         return out + "_0.npy"
+    @staticmethod
+    def pad_dataset(df, raw_dataset_path: str, raw_label_path: str, min_amount_of_pictures=3):
+        run_correction = False
+
+        values = df['species'].value_counts().keys().tolist()
+        counts = df['species'].value_counts().tolist()
+
+        itt = 0
+
+        less_than_list = []
+
+        for count in counts:
+            if min_amount_of_pictures > count:
+                less_than_list.append((values[itt], count))
+                run_correction = True
+            itt += 1
+
+        if run_correction:
+            world_df: pd.DataFrame = pd.read_csv(raw_dataset_path, sep="	", low_memory=False)
+            world_df_labels: pd.DataFrame = pd.read_csv(raw_label_path, sep=",", low_memory=False)
+
+            Database.drop_cols([world_df, world_df_labels])
+
+            world_df = world_df.merge(world_df_labels[world_df_labels['gbifID'].isin(world_df['gbifID'])],
+                                      on=['gbifID'])
+            # print("world_df: ", world_df.columns)
+
+            out = df["species"].value_counts()
+
+            species_with_less_than_optimal_amount_of_images = []
+
+            totalRows = 0
+            for index, count in out.items():
+                if count < min_amount_of_pictures:
+                    species_with_less_than_optimal_amount_of_images.append(index)
+                    totalRows += 1
+
+            totalRows = totalRows * min_amount_of_pictures
+            print("Total number of extra rows: ", totalRows)
+            world_df = world_df.loc[world_df['species'].isin(species_with_less_than_optimal_amount_of_images)]
+            # loop that gets the species, which are below the required amount
+            for item, count in less_than_list:
+                # print("World pictures to add: ", min_amount_of_pictures - count)
+                world_specific = world_df.loc[world_df["species"] == item]
+                world_specific = world_specific.iloc[:min_amount_of_pictures - count]
+                df = df.append(world_specific, ignore_index=True)
```

Comment:
Er det meningen, at der skal bruges `df.append` her?

Reply:
dataframe.append is deprecated in numpy 1.24.3